h5io/_h5io.py

```diff
@@ -423,3 +424,63 @@ def __init__(self):
 
     def __del__(self):
         rmtree(self._path, ignore_errors=True)
+
+
+def _list_file_contents(h5file):
+    h5py = _check_h5py()
```

Comment:
as currently implemented it does because the public function quick checks whether it needs to do a `with` statement. I could add the check for `File` type in the public function and then remove the import from the hidden function